### ISSUE TITLE
Temporary files are permanent with standalone spark cluster

### ISSUE THREAD

**Issue by r0mainK** [src-d/engine#348]
## Expected Behavior

After a spark job using the engine to process data completes, the engine should remove all temporary fields it created. Ideally, inputs should be removed after each spark task ends.

## Current Behavior
When running jobs on the Kubernetes cluster, for example ml's `repos2ids`, if we use spark in standalone mode then whether a job fails or not, all temporary data created in the worker pods remains. This is a definitive blocker when trying to run jobs on large inputs. This does not happen when running the job with the local spark.

## Possible Solution

When using spark in standalone the engine must do cleaning up on the workers, if possible between spark tasks (if we have to wait for the job to finish when processing PGA for it to start cleaning up, we will need a lot more disk memory)

## Steps to Reproduce (for bugs)

- Connect to cluster and launch a pod:
```
kubectl config use-context local-gpu
kubectl run -i --tty engine-test --image=srcd/ml-core --generator="run-pod/v1" --command -- /bin/bash
```
- Update dependencies, then launch job locally to check the temporary files are collected:
```
pip3 uninstall sourced-engine -y  # 0.5.3 is installed by defaut atm
pip3 install sourced-engine sourced-ml -y
srcml repos2ids -r hdfs://hdfs-namenode/5-siva -o here_out --dep-zip --conf spark.local.dir=/spark-temp-data
du -h /spark-temp-data
```
- Launch job with standalone: `srcml repos2ids -r hdfs://hdfs-namenode/5-siva -o here_out --dep-zip --conf spark.local.dir=/spark-temp-data -s spark://l-spark-spark-master:7077`
- Connect to any worker pod: `kubectl exec -it l-spark-spark-worker-<ID> bash`
-temp files should still be there: `du -h /spark-temp-data`

## Context

This has affected me, as well as all member of the ml team that tried launching jobs using `engine` to process large inputs on `local-gpu` or `staging` (in my case `apollo`).  As a consequence of this issue, temporary data fills up the worker pods, until the Kubernetes controller senses disk pressure on one or multiple nodes, kills the workers and causes our jobs to fail. 
To give a sense of the impact of this, we have not been able to process files larger then 20G on the cluster atm.

For more info, check out problem 2 of [this google doc](https://docs.google.com/document/d/1M2nHwCagndw0h9-i7h_Gwl9kISoOuIX5J_yXjYOuvZI/edit?ts=5ab25fd2).

## Your Environment (for bugs)
* Version used: last, but this was happenning since at least 0.5.2
* Operating System and version: pods on the cluster so Ubuntu 16.04 I believe
* Some needed resources to reproduce the problem: Access to one of the clusters.

**Comment by r0mainK** [comment]
@erizocosmico sorry for the delay, I launched a dozen jobs yesterday and used your solution so that it would clean every 30min, but it did not work

**Comment by erizocosmico** [comment]
I've been trying a few things but in standalone mode, at least in the pipeline-staging cluster it deletes the files for me.

This is the folder after running a job:

```
<park-25bffdaa-07ed-48a6-a22f-076d1044e225# du -h
0	./processing-repositories
664K	./siva-files
664K	.
```

As you can see all files have been deleted (except for the folders themselves).

Right now, `processing-repositories` also keeps another level of folders, but I will PR a fix for that. In any case, the uncompressed files are deleted.

Do you have by any chance the logs of the worker where you executed this? Is it possible that you're running this with  `skipCleanup`?

**Comment by r0mainK** [comment]
Okay so after seeing this I've been trying stuff on local-gpu.

Regarding `skipCleanup`, I saw today that when using ml default was to set it to true: https://github.com/src-d/ml/pull/237 (this is changed in the PR)

I ran a test job after setting this value to `false` and it seems that in Cluster mode it also has the same behaviour you saw, folders remain but content is deleted.

**Comment by erizocosmico** [comment]
Ok, that was the issue then. In any case, I found cases where a few repositories could be left and extra folders were being kept. All that is fixed in #380

**Comment by r0mainK** [comment]
Awesome thanks a lot ^^ will close the issue when the PR is merged then.
